### PR TITLE
fix issues rendering soft set hooks

### LINKED ISSUE #9: All attribute hooks are not rendered
I'm having some problems with value attribute being rendered as [object Object] instead of its actual value. This is because a soft hook is created and the following code does not render it because of the second condition `value.namespace`.

```javascript
if (isVHook(value) && value.namespace) {
   ret += ' ' + createAttribute(name, value.value, true);
   continue;
}
```

As an example this test fails if added:

```javascript
it('should render hooks', function () {
  var node = h('input', { type: 'submit', value: 'add' });
  assert.equal(toHTML(node), '<input type="submit" value="add">');
});
```

### PR DESCRIPTION
**Pull request by bitinn**
fix #9 

I don't see a better way to distinguish these 3 hooks with `is-vhook` alone.

https://github.com/Matt-Esch/virtual-dom/blob/master/virtual-hyperscript/hooks/attribute-hook.js
https://github.com/Matt-Esch/virtual-dom/blob/master/virtual-hyperscript/hooks/ev-hook.js
https://github.com/Matt-Esch/virtual-dom/blob/master/virtual-hyperscript/hooks/soft-set-hook.js

### CODE REVIEW

**Comment by nthtran** [comment]
Yeah don't see another way. Thanks!

**Comment by marlun** [comment]
All this time trying to figure out the best (an easy) way to try a pull-request of an installed npm package and now I can just run `npm update vdom-to-html` :smile:

Thanks guys, works now!

I also checked how virtual-dom-stringify solved it

```javascript
if (typeof attrVal === 'object' && key !== 'style') {
  attrVal = attrVal.value;
}
```

File: index.js [hunk 1]
```diff
@@ -52,7 +53,7 @@ function openTag(node) {
       value = css.trim();
     }
 
-    if (isVHook(value) && value.namespace) {
+    if (value instanceof softHook || value instanceof attrHook) {
```
Comment (DeTeam) [inline index.js]: Guess it's too late, but stil.
@nthtran do you think it make sense to have `instanceof` check here?

Problem is that instances of hooks used in vdom-to-html and users' libs can be different, they will point to different constructors of SoftHook and AttrHook.

Comment (DeTeam) [inline index.js]: Mb peerDependencies would make sense?

**Pull request merged**